packages/reporting/src/components/report-dashboard.tsx

```diff
@@ -95,6 +97,71 @@ export const ReportDashboard = (props: ReportDashboardProps) => {
   const [currentReport, setCurrentReport] = React.useState(Reports.queryAllLogs);
   const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
 
+  const [fromLogDate, setFromLogDate] = React.useState<MaterialUiPickersDate>(new Date());
+  const [toLogDate, setToLogDate] = React.useState<MaterialUiPickersDate>(new Date());
+
+  const [header, setHeader] = React.useState<string>('');
+
+  const handleFromLogDateChange = React.useCallback((date: MaterialUiPickersDate) => {
+    setFromLogDate(date);
+  }, []);
+
+  const handleToLogDateChange = React.useCallback((date: MaterialUiPickersDate) => {
+    setToLogDate(date);
+  }, []);
+
+  const itemConfig = (props: ConfigProps): JSX.Element | null => {
+    if (Object.prototype.hasOwnProperty.call(reportContainer, currentReport)) {
+      switch (currentReport) {
+        case Reports.queryAllLogs:
+          const QueryConfig = reportContainer[currentReport];
+          return <QueryConfig />;
+        case Reports.showDispenserStateReport:
+        case Reports.showDoorStateReport:
+        case Reports.showFleetStateReport:
+        case Reports.showHealthReport:
+        case Reports.showIngestorStateReport:
+        case Reports.showLiftStateReport:
+        case Reports.showLoginsReport:
+        case Reports.showLogoutsReport:
+        case Reports.showLoginFailuresReport:
+        case Reports.showTasksReport:
+          const Config = reportContainer[currentReport];
+          return <Config {...props} />;
+        default:
+          return null;
+      }
+    } else {
+      return null;
+    }
+  };
+
+  const headerTitle = (): void => {
+    switch (currentReport) {
+      case Reports.queryAllLogs:
+        setHeader(currentReport.split('query')[1]);
+        break;
+      case Reports.showDispenserStateReport:
+      case Reports.showDoorStateReport:
+      case Reports.showFleetStateReport:
+      case Reports.showHealthReport:
+      case Reports.showIngestorStateReport:
+      case Reports.showLiftStateReport:
+      case Reports.showLoginsReport:
+      case Reports.showLogoutsReport:
+      case Reports.showLoginFailuresReport:
+      case Reports.showTasksReport:
+        setHeader(currentReport.split('show')[1].split('Report')[0]);
+        break;
+      default:
+        break;
+    }
+  };
+
+  React.useEffect(() => {
```

Comment:
I think it is needed because without `useEffect`, it will cause the component to re-render infinitely and crash it.

Reply:
It shouldn't matter, unless react has different logic when it comes to updating a state during render and in the effect?

An effect without a dependency would run everytime, so after each render, it would call `headerTitle()` which would set the state, resulting in the same infinite loop if the state is different everytime. If `setHeader` results in the same string, it shouldn't cause a re-render in either cases.

If it did turns out react have different logic in handling state updates in renders vs effects, then I think you should use `useMemo`, it should be slightly more efficient and "correct".